Library/Homebrew/cmd/update-report.rb

```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
       report_all = false
     end
 
-    dump_new_formula_report
-    dump_new_cask_report
+    dump_new_formula_report if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE"].blank?
```

Comment:
Define this in `env_config.rb` instead and use the generated `Homebrew.no_update_report*?` methods